dev/io.openliberty.mcp/src/io/openliberty/mcp/messaging/Cancellation.java

```diff
@@ -73,4 +73,9 @@ class OperationCancellationException extends RuntimeException {
 
     }
 
+    /**
+     * Cancels the ongoing operation associated with this instance.
+     */
+    void cancel();
+
```

Comment:
This method should not be on the interface, it should only be on `CancellationImpl` so that only we can call it.

This interface is passed to the tool method and we don't allow the tool method to cancel the request.